src/main/java/org/mule/tools/apikit/output/MuleConfigGenerator.java

```diff
@@ -230,12 +238,141 @@ private MuleConfig getUpdatedMuleConfig(MuleConfig muleConfig, Element newApikit
+    MuleConfig global = createCommonPropertiesFile();
+    ConfigurationPropertiesConfig configurationPropertiesConfig =
+        configuration.getScaffoldingAccessories().getProperties() != null ? createConfigurationProperties() : null;
     for (ApikitMainFlowContainer api : apiContainers) {
-      muleConfigs.add(api.getMuleConfig() == null ? createMuleConfig(api) : api.getMuleConfig());
+      muleConfigs.addAll(processMuleConfig(global, configurationPropertiesConfig, api));
     }
     return muleConfigs;
   }
 
+  private Set<MuleConfig> processMuleConfig(MuleConfig global, ConfigurationPropertiesConfig configurationPropertiesConfig,
+                                            ApikitMainFlowContainer api) {
+    Set<MuleConfig> muleConfigs = new HashSet<>();
+    FlowScope flowScope = null;
+    String muleConfigID = createMuleConfigID(api.getId());
+    String mainFlowRef = FilenameUtils.removeExtension(muleConfigID).concat(MAIN_FLOW_SUFFIX);
+    APIAutodiscoveryConfig apiAutodiscoveryConfig = createAPIAutodiscoveryConfig(mainFlowRef);
+    MuleConfig muleConfig = createMuleConfig(api);
+    if (global != null) {
+      commonConfigurations(api, global);
+      global = addCustomizations(configurationPropertiesConfig, apiAutodiscoveryConfig, global);
+      flowScope = new FlowScope(api, isMuleEE(), global.getApikitConfigs().stream().findFirst().orElse(null).getName());
+      addMuleConfig(muleConfigs, fromDoc(global.buildContent()),
+                    configuration.getScaffoldingAccessories().getExternalConfigurationFile());
+    } else {
+      muleConfig = addCustomizations(configurationPropertiesConfig, apiAutodiscoveryConfig, muleConfig);
+    }
+    if (api.getMuleConfig() == null) {
+      flowScope = flowScope == null ? new FlowScope(api, isMuleEE()) : flowScope;
+      muleConfig.addFlow(new Flow(flowScope.generate()));
+      addConsoleFlow(api, muleConfig);
+      muleConfig = fromDoc(muleConfig.buildContent());
+      api.setMuleConfig(muleConfig);
+    } else {
+      muleConfigID = api.getMuleConfig().getName();
+    }
+    addMuleConfig(muleConfigs, muleConfig, muleConfigID);
+    return muleConfigs;
+  }
+
+  /**
+   * Used for adding custom attributes that the mule config may have
+   *
+   * @param apiAutodiscoveryConfig        API Autodiscovery configuratoin to add.
+   * @param configurationPropertiesConfig Configuration properties configuration to add.
+   * @param muleConfig                    Mule configuration to customize.
+   * @return muleConfig with customizations.
+   */
+  private MuleConfig addCustomizations(ConfigurationPropertiesConfig configurationPropertiesConfig,
+                                       APIAutodiscoveryConfig apiAutodiscoveryConfig, MuleConfig muleConfig) {
+    muleConfig.setConfigurationPropertiesConfig(configurationPropertiesConfig);
+    muleConfig = setAPIAutodiscoveryId(muleConfig, apiAutodiscoveryConfig);
+    return muleConfig;
```

Comment:
why setConfigurationPropertiesConfig and after overwrite the value muleConfig = setAPIAutodiscoveryId(muleConfig, apiAutodiscoveryConfig); it should be consistent